kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -24,42 +24,100 @@
 namespace Kratos
 {
 	CalculateDiscontinuousDistanceToSkinProcess::CalculateDiscontinuousDistanceToSkinProcess(ModelPart& rVolumePart, ModelPart& rSkinPart)
-	: FindIntersectedGeometricalObjectsProcess(rVolumePart, rSkinPart) {
-		
+		: mrSkinPart(rSkinPart), mrVolumePart(rVolumePart), mFindIntersectedObjectsProcess(rVolumePart, rSkinPart)
+	{
+	}
+
+	CalculateDiscontinuousDistanceToSkinProcess::~CalculateDiscontinuousDistanceToSkinProcess()
+	{
 	}
 
-	CalculateDiscontinuousDistanceToSkinProcess::~CalculateDiscontinuousDistanceToSkinProcess(){}
+	void CalculateDiscontinuousDistanceToSkinProcess::Initialize()
+	{
+		// Initialize the intersected objects process
+		mFindIntersectedObjectsProcess.Initialize();
 
+		// Reset the nodal distance values
+		const double initial_distance = 1.0;
```

Comment:
Good question, the point is to avoid an excessively large value to avoid the problems we detect a few weeks ago. Since the bounding box size is not expected to be excessively large, I agree.